tests/darwin/importer/formats/import_dataloop_test.py

```diff
@@ -0,0 +1,138 @@
+
+    DARWIN_PARSED_DATA = {
+        "filename": "test.jpg",
+        "annotations": [
+            {"class": "class_1"},
+            {"class": "class_2"},
+            {"class": "class_3"},
+        ],
+    }
+
+
+class TestParsePath(DataLoopTestCase):
+    def tearDown(self):
+        patch.stopall()
+
+    @patch(
+        "darwin.importer.formats.dataloop._remove_leading_slash",
+    )
+    def test_returns_none_if_file_extension_is_not_json(self, mock_remove_leading_slash):
+        self.assertIsNone(parse_path(Path("foo.bar")))
+
+    @patch(
+        "darwin.importer.formats.dataloop._remove_leading_slash",
+    )
+    @patch("darwin.importer.formats.dataloop.json.load")
+    @patch("darwin.importer.formats.dataloop.Path.open")
+    @patch("darwin.importer.formats.dataloop._parse_annotation")
+    def test_opens_annotations_file_and_parses(
+        self,
+        _parse_annotation_mock: MagicMock,
+        path_open_mock: MagicMock,
+        json_load_mock: MagicMock,
+        mock_remove_leading_slash: MagicMock,
+    ):
+        json_load_mock.return_value = self.DARWIN_PARSED_DATA
+        test_path = "foo.json"
+
+        parse_path(Path(test_path))
+
+        self.assertEqual(_parse_annotation_mock.call_count, 3)
+        path_open_mock.assert_called_once()
+        json_load_mock.assert_called_once()
+        mock_remove_leading_slash.assert_called_once()
+
+
+class TestRemoveLeadingSlash(DataLoopTestCase):
+    def tearDown(self) -> None:
+        patch.stopall()
+
+    def test_removes_slash_if_present(self):
+        self.assertEqual(_remove_leading_slash("/foo"), "foo")
+
+    def test_does_not_remove_slash_if_not_present(self):
+        self.assertEqual(_remove_leading_slash("foo"), "foo")
+
+
+class TestParseAnnotation(DataLoopTestCase):
+    def setUp(self):
+        super().setUp()
+        self.parsed_json = json_loads(self.DATALOOP_MOCK_DATA)
+
+    def tearDown(self) -> None:
+        patch.stopall()
+
+    def test_handles_box_type(self):
+        from darwin.importer.formats.dataloop import _parse_annotation as pa
+
+        with patch("darwin.importer.formats.dataloop.dt.make_bounding_box") as make_bounding_box_mock:
+            make_bounding_box_mock.return_value = Annotation("class_1", 0, 0, 0, 0)
+            pa(self.parsed_json["annotations"][0])  # 0 is a box type
+
+            make_bounding_box_mock.assert_called_with("box_class", 288.81, 845.49, 1932.5100000000002, 2682.75)
+
+    def test_handles_class_type(self):
+        annotation = _parse_annotation(self.parsed_json["annotations"][1])  # 1 is a class type
+        self.assertEqual(annotation, None)
+
+    def test_handles_segment_type(self):
+        from darwin.importer.formats.dataloop import _parse_annotation as pa
+
+        with patch("darwin.importer.formats.dataloop.dt.make_polygon") as make_polygon_mock:
+            pa(self.parsed_json["annotations"][2])  # 2 is a segment type
+
+            if "kwargs" in make_polygon_mock.call_args:
```

Comment:
`MagicMock` and `Mock` before version 3.8 don't support testing `kwargs`, so testing this would be difficult, but this feature switch will mean we still test against 3.8 and 3.9 in both Windows and Ubuntu, which I think is plenty for a simple dict comprehension.